core/src/shared_memory.rs

```diff
@@ -1,36 +1,45 @@
 use crate::syscalls;
+use core::cell::UnsafeCell;
 use core::ptr;
 
 #[must_use = "Shared memory risks being dropped too early. Drop it manually."]
-pub struct SharedMemory<'a> {
+pub struct SharedMemory<T> {
     driver_number: usize,
     allow_number: usize,
-    buffer_to_share: &'a mut [u8],
+    buffer_to_share: UnsafeCell<T>,
```

Comment:
I think this is wrong! Instead of enforcing to pass an array reference which would pin the referenced array it is now possible to pass an entire array. This should result in UB once you move the `SharedMemory` object around.

Why not change the signature to `UnsafeCell<&'a mut [u8]>`?

Reply:
Good point, I fixed this.